lib/action_subscriber.rb

```diff
@@ -135,5 +139,6 @@ def self.default_routes
 require "action_subscriber/railtie" if defined?(Rails)
 
 at_exit do
+  ::ActionSubscriber::Publisher::Async.publisher_adapter.shutdown!
```

Comment:
I'm not sure that I love using `at_exit` to drain the pool, but it will terminate after `seconds_to_wait_for_graceful_shutdown` has expired with nice messages.